src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,790 @@
+<-trans {1+[2 x ]} {1+[2 y ]} {2[1+ .y ]} (odd<odd x<y) (odd<even (inj₂ refl)) =
+                                                             odd<even (inj₁ x<y)
+<-trans {1+[2 x ]} {1+[2 y ]} {1+[2 z ]} (odd<odd x<y) (odd<odd y<z) =
+                                                       odd<odd (<-trans x<y y<z)
+
+<⇒≯ : _<_ ⇒ _≯_
+<⇒≯ x<y y<x =  <-irrefl refl (<-trans x<y y<x)
+
+>⇒≮ : _>_ ⇒ _≮_
+>⇒≮ y<x =  <⇒≯ y<x
+
+<⇒≤ : _<_ ⇒ _≤_
+<⇒≤ = inj₁
+
+------------------------------------------------------------------------------
+-- resolving inequality comparison
+------------------------------------------------------------------------------
+
+open Tri
+
+<-cmp :  ∀ (x y) → Tri (x < y) (x ≡ y) (x > y)
+<-cmp zero     zero      = tri≈ ≮0     refl  ≮0
+<-cmp zero     2[1+ _ ]  = tri< 0<even (λ()) ≮0
+<-cmp zero     1+[2 _ ]  = tri< 0<odd  (λ()) ≮0
+<-cmp 2[1+ _ ] zero      = tri> (λ())  (λ()) 0<even
+<-cmp 2[1+ x ] 2[1+ y ]  with <-cmp x y
+... | tri< x<y _   _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)  where lt = even<even x<y
+... | tri≈ _   x≡y _ =  tri≈ (<-irrefl eq) eq (<-irrefl (sym eq))
+  where
+  eq = cong 2[1+_] x≡y
+... | tri> _ _ x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt : 2[1+ x ] > 2[1+ y ]
+  gt = even<even x>y
+
+<-cmp 2[1+ x ] 1+[2 y ]  with <-cmp x y
+... | tri< x<y _    _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)
+  where
+  lt = even<odd x<y
+
+... | tri≈ _   x≡y _ =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  1+2x≡1+2y = cong 1+[2_] x≡y;  1+2x<2[1+x] = 1+[2x]<2[1+x] x
+
+  gt : 1+[2 y ] < 2[1+ x ]
+  gt = subst (_< 2[1+ x ]) 1+2x≡1+2y (1+[2x]<2[1+x] x)
+
+... | tri> _ _ y<x =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt : 1+[2 y ] < 2[1+ x ]
+  gt = odd<even (inj₁ y<x)
+
+<-cmp 1+[2 _ ] zero =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt = 0<odd
+
+<-cmp 1+[2 x ] 2[1+ y ]  with <-cmp x y
+... | tri< x<y _ _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)
+  where
+  lt = odd<even (inj₁ x<y)
+
+... | tri≈ _ x≡y _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)
+  where
+  lt = odd<even (inj₂ x≡y)
+
+... | tri> _ _ x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt = even<odd x>y
+
+<-cmp 1+[2 x ] 1+[2 y ]  with <-cmp x y
+... | tri< x<y _ _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)
+  where
+  lt = odd<odd x<y
+
+... | tri≈ _ x≡y _ =  tri≈ (≡⇒≮ eq) eq (≡⇒≯ eq)
+  where
+  eq = cong 1+[2_] x≡y
+
+... | tri> _ _ x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt = odd<odd x>y
+
+------------------------------------------------------------------------------
+-- Several monotonicity proofs for toℕ, fromℕ.
+------------------------------------------------------------------------------
+
+toℕ-mono-< :  toℕ Preserves _<_ ⟶ ℕ._<_
+toℕ-mono-< {_}        {zero}     ()
+toℕ-mono-< {zero}     {2[1+ _ ]} _               =  ℕₚ.0<1+n
+toℕ-mono-< {zero}     {1+[2 _ ]} _               =  ℕₚ.0<1+n
+toℕ-mono-< {2[1+ x ]} {2[1+ y ]} (even<even x<y) =  x'N<y'N
+  where
+  open ℕₚ.≤-Reasoning
+  xN      = toℕ x
+  yN      = toℕ y
+  xN<yN   = toℕ-mono-< x<y
+  x'N<y'N = begin
```

Comment:
This reasoning proof doesn't need to be in the `where`. Same with many examples below.

Reply:
Fixed.